src/main/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtil.java

```diff
@@ -418,6 +422,54 @@ public static ResponseEntity<Object> setOrgIdInGetUserResponse(ResponseEntity<Ob
         return newResponseEntity;
     }
 
+    public static GetRefreshUsersResponse buildGetRefreshUsersResponse(Page<ProfessionalUser> professionalUsersPage,
+                                                                   List<ProfessionalUser> professionalUsers,
+                                                                   List<UserConfiguredAccess> userConfiguredAccesses) {
+        GetRefreshUsersResponse getRefreshUsersResponse = new GetRefreshUsersResponse();
+        getRefreshUsersResponse.setMoreAvailable(professionalUsersPage != null && !professionalUsersPage.isLast());
+        List<RefreshUser> refreshUserList = new ArrayList<>();
+
+        for (ProfessionalUser professionalUser : professionalUsers) {
+            RefreshUser refreshUser = new RefreshUser();
+            refreshUser.setUserIdentifier(professionalUser.getUserIdentifier());
+            refreshUser.setLastUpdated(professionalUser.getLastUpdated());
+            refreshUser.setOrganisationIdentifier(professionalUser.getOrganisation().getOrganisationIdentifier());
+
+            List<UserAccessType> userAccessTypes = new ArrayList<>();
+
+            userAccessTypes.addAll(userConfiguredAccesses.stream()
+                    .filter(uca -> uca.getUserConfiguredAccessId().getProfessionalUser().getUserIdentifier()
+                            .equals(professionalUser.getUserIdentifier()))
+                    .map(RefDataUtil::fromUserConfiguredAccess)
+                    .toList());
+
+            refreshUser.setUserAccessTypes(userAccessTypes);

```

Comment:
can we use constructor instead of setter methods on refreshUser

new RefreshUser(professionalUser.getUserIdentifier(), professionalUser.getLastUpdated(), professionalUser.getOrganisation().getOrganisationIdentifier(), userAccessTypes)

Reply:
refactored to use create obj using constructor in [eb360dc](https://github.com/hmcts/rd-professional-api/pull/1529/commits/eb360dc1e9f8405d50ffea39011aa6a4e03c82a1)